src/Mod/Part/App/TopoShapeExpansion.cpp

```diff
@@ -509,6 +514,743 @@ void TopoShape::mapSubElement(const std::vector<TopoShape>& shapes, const char*
     }
 }
 
+struct ShapeInfo
+{
+    const TopoDS_Shape& shape;
+    TopoShapeCache::Ancestry& cache;
+    TopAbs_ShapeEnum type;
+    const char* shapetype;
+
+    ShapeInfo(const TopoDS_Shape& shape, TopAbs_ShapeEnum type, TopoShapeCache::Ancestry& cache)
+        : shape(shape)
+        , cache(cache)
+        , type(type)
+        , shapetype(TopoShape::shapeName(type).c_str())
+    {}
+
+    int count() const
+    {
+        return cache.count();
+    }
+
+    TopoDS_Shape find(int index)
+    {
+        return cache.find(shape, index);
+    }
+
+    int find(const TopoDS_Shape& subshape)
+    {
+        return cache.find(shape, subshape);
+    }
+};
+
+////////////////////////////////////////
+// makESHAPE -> makeShapeWithElementMap
+///////////////////////////////////////
```

Comment:
Is this ready to be removed?

Reply:
I was actually planning on leaving it until we were basically done, as a sort of breadcrumb to search for. I could be dissuaded though.

Reply:
Nope, that's fine!